test/unit/expiration_process_test.lua

```diff
@@ -136,3 +136,37 @@ function g.test_execution_error(cg)
     t.assert_equals(task.restarts, 1)
     task:kill()
 end
+
+function g.test_check_tuples_not_expired_by_timestamp(cg)
```

Comment:
As I understand it, the test verifies that nothing will break if for some time `is_tuple_expired` returns false (several iterations will be traversed), and only then it will start returning true and all tuples are archived correctly.